system/lib/pthread/proxying.c

```diff
@@ -583,31 +632,35 @@ em_promise_t emscripten_proxy_promise(em_proxying_queue* q,
                           &block->promise_ctx);
 }
 
-typedef struct proxied_js_func_t {
-  int funcIndex;
-  void* emAsmAddr;
-  pthread_t callingThread;
-  int numArgs;
-  double* argBuffer;
-  double result;
-  bool owned;
-} proxied_js_func_t;
-
 static void run_js_func(void* arg) {
   proxied_js_func_t* f = (proxied_js_func_t*)arg;
   f->result = _emscripten_receive_on_main_thread_js(
-    f->funcIndex, f->emAsmAddr, f->callingThread, f->numArgs, f->argBuffer);
+    f->funcIndex, f->emAsmAddr, f->callingThread, f->numArgs, f->argBuffer, NULL);
   if (f->owned) {
     free(f->argBuffer);
     free(f);
   }
 }
 
+static void run_js_promise_await(void* arg) {
+  proxied_js_func_t* f = (proxied_js_func_t*)arg;
+  f->result = _emscripten_receive_on_main_thread_js(
+    f->funcIndex, f->emAsmAddr, f->callingThread, f->numArgs, f->argBuffer, (void*)f->ctx);
+}
+
+void _emscripten_proxy_promise_finish(void* res, em_proxying_ctx* ctx) {
```

Comment:
I would expect the context to be the first arugment here.

Reply:
👍